test/jdk/java/awt/event/MouseEvent/ResizeMouseExitEnterMisfire/ResizeMouseExitEnterMisfire.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.SwingUtilities;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+/**
+ * @test
+ * @bug 8254841
+ * @summary Tests spurious firing of mouseEntered and mouseExit on resize
+ * @run main ResizeMouseExitEnterMisfire
+ */
+public class ResizeMouseExitEnterMisfire {
+
+    // Keep track of number of mouseEntered and mouseExited events
+    private static volatile int mouseEntered = 0;
+    private static volatile int mouseExited = 0;
+
+    JFrame frame;
+    JLabel label;
+    int xLocInnerCorner;
+    int yLocInnerCorner;
+    int xLocOuterCorner;
+    int yLocOuterCorner;
+    int xLocNewEdge;
+    int yLocNewEdge;
+
+    Robot robot;
+
+    public void createAndShowFrame() throws Exception {
+        SwingUtilities.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                frame = new JFrame();
+                frame.setSize(200, 200);
```

Comment:
It is preferred to move the frame to centre of screen by calling frame.setLocationRelativeTo(null)

Reply:
I use the frame size to find the correct location of the mouse for robot, so moving the frame to the center of the screen would make calculating mouse locations more complicated...
And I need some space to expand the window, which could be limited if the frame were to be in the center, so in this case it's probably better to keep the window in the top left corner? Unless there's a reasoning I missed for moving it to the center?

Reply:
This is done because the Robot sometime ends up clicking on Taskbar or side bar on Linux and we have had some related issue. Moving to centre helps to avoid those issues. Also, you just need to call frame.getLocationOnScreen to get the location along with the frame.bounds to get correct location. But this all is optional/preferred and it is fine you would like to keep the frame in left top.

Reply:
I see, let me make that change then